pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -476,6 +476,16 @@ func getStsAnnotations(val *Values, sts *appsv1.StatefulSet) map[string]string {
 		},
 		val.Annotations,
 	)
+
+	// If `scaleToMultiNodeAnnotationKey` annotation is already present in etcd statefulset
+	// then it is better to check scale-up was successful or not before removing `scaleToMultiNodeAnnotationKey` annotation.
+	if metav1.HasAnnotation(sts.ObjectMeta, scaleToMultiNodeAnnotationKey) {
```

Comment:
There is a chance that the sts could be nil due to a [`getExistingSts()`](https://github.com/gardener/etcd-druid/blob/master/pkg/component/etcd/statefulset/statefulset.go#L155) call and could result in a nil pointer dereference
Please add a `sts != nil` check here